probe_src/probe_py/test_probe.py

```diff
@@ -0,0 +1,253 @@
+from typer.testing import CliRunner
+import tarfile
+from .cli import app
+from . import parse_probe_log
+from . import analysis
+import pathlib
+import typing
+import networkx as nx
+import subprocess
+
+runner = CliRunner()
+
+def test_diff_cmd():
+    result = runner.invoke(app,["record","--make","diff","../flake.nix","../flake.lock"])
```

Comment:
I know I said to do `typer.CliRunner`, but a new complication arose: I want to be able to use this test suite with the Rust CLI front-end, so we should switch this away from using `typer.CliRunner`
```suggestion
    subprocess.run(["./PROBE", "record", "diff", "../flake.nix", "flake.lock"], check=True, capture_output=True)
```

Reply:
I tried making this change. I am stuck at this error 
```
stderr = 'Traceback (most recent call last):\n  File "<frozen runpy>", line 198, in _run_module_as_main\n  File "<frozen runpy>...i.py", line 5, in <module>\n    import typing_extensions\nModuleNotFoundError: No module named \'typing_extensions\'\n'
retcode = 1
```
It seems the tests are not executing in the nix environment. I tried the following solutions:
-  using the OS environment variables while executing `subprocess.run` 
- including `typing-extensions` in flake.nix
- adding and removing few environment variables
It's going to take some time to resolve this. I tried directly using the `record` function, it works fine! @charmoniumQ

Reply:
This snippet will work if you run pytest like `python -m pytest <pytest_args...>` rather than `pytest <pytest_args...>`. The difference is that the former adds Pytest's copy of Python to the `$PATH` and the latter does not.

@saleha-muzammil we should make sure the Justfile does `python -m pytest ...` rather than `pytest ...`

Reply:
And we should be doing `subprocess.run` instead of `record` because in the future, the performance-critical parts of PROBE will be implemented in Rust (see #20) and `subprocess.run` will work will with _either_ Rust or Python (and it's closer to the user's actual interface).

But I'm glad you could test the tests with `record(...)` before I debugged `subprocess.run(...)`. That's a good workaround :)

Reply:
Yup! 
I am resolving the issue with subprocess.run, I'll change the code as soon as it is resolved.

Reply:
> This snippet will work if you run pytest like `python -m pytest <pytest_args...>` rather than `pytest <pytest_args...>`. The difference is that the former adds Pytest's copy of Python to the `$PATH` and the latter does not.
> 
> @saleha-muzammil we should make sure the Justfile does `python -m pytest ...` rather than `pytest ...`

@charmoniumQ updated the justfile accordingly